server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/LongKeyedBucketOrds.java

```diff
@@ -274,4 +311,129 @@ public void close() {
             ords.close();
         }
     }
+
+    public static class FromManySmall extends LongKeyedBucketOrds {
+        private final LongHash ords;
+        private final int owningBucketOrdShift;
+        private final long owningBucketOrdMask;
+
+        public FromManySmall(BigArrays bigArrays, int owningBucketOrdShift) {
+            ords = new LongHash(2, bigArrays);
+            this.owningBucketOrdShift = owningBucketOrdShift;
+            this.owningBucketOrdMask = -1L << owningBucketOrdShift;
+        }
+
+        private long encode(long owningBucketOrd, long value) {
+            // This is in the critical path for collecting lots of aggs. Be careful of performance.
```

Comment:
This isn't 100% true - it's only the hot path for a `terms` agg under an agg that knows its cardinality like `range` or `filters`. Or a `date_histogram` when it's been rewritten into `range` or `filters`.

Reply:
Yeah. I guess the untrue bit is "lots". I'll just remove that word.